react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+class ValidateInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entry: 'start',
+      userPassword: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.NoMoreRepeatedCode = this.NoMoreRepeatedCode.bind(this);
+    this.ConditionalStatuses = this.ConditionalStatuses.bind(this);
+    this.ConditionalImages = this.ConditionalImages.bind(this);
+  }
+
+  handleSubmit(event) {
+    this.setState({
+      userPassword: event.target.value
+    });
+    event.preventDefault();
+  }
+
+  handleChange(event) {
+    if (event.target.value.length >= 8) {
+      this.setState({
+        entry: 'pass'
+      });
+    } else if (event.target.value.length === 0) {
+      this.setState({
+        entry: 'empty'
+      });
+    } else if (event.target.value.length < 8) {
+      this.setState({
+        entry: 'short'
+      });
+    }
+  }
+
+  NoMoreRepeatedCode() {
+    return (
+      <form className="container" onSubmit={this.handleSubmit}>
+        <label>Password</label>
+        <div className="input-container">
+          <input type="password" className="input" value={this.state.handleSubmit} onChange={this.handleChange}/>
```

Comment:
The `value` prop should be something like `this.state.userPassword`. and `handleChange` should update the state with every keystroke.